runtime/src/payout.rs

```diff
@@ -0,0 +1,208 @@
+use super::*;
+pub use sp_runtime::{Percent, Perquintill};
+
+fn abs(v: f64) -> f64 {
+    if v > 0f64 {
+        v
+    } else {
+        -v
+    }
+}
+
+fn exp(p: f64) -> f64 {
+    let mut res: f64 = 0f64;
+    // Sums: 1 + x + x^2/2! + x^3/3! + x^4/4! + x^5/5! +
+    let mut next = 1f64;
+    let mut i = 1f64;
+    while abs(next) > ExpPrecision::get() {
+        res += next;
+        next = (next * p) / i;
+        i += 1f64;
+    }
+    res
+}
+
+fn to_inflation(p: f64) -> Perquintill {
+    Perquintill::from_rational(
+        (exp(p) * (0.01f64 / ExpPrecision::get())) as u64,
+        (1f64 / ExpPrecision::get()) as u64,
+    )
+}
+
+/// Implements a custom inflation model based on the following formula:
+/// I(s_c) = I_b + I_v(s_c)
+///
+/// with:
+/// I_v(x) = C * exp(K * (1 - (s_c / s_t)))
+///
+/// where:
+/// I_b: base inflation
+/// I_v: variable inflation
+/// s_t: target inflation
```

Comment:
that's target _staking_ and not inflaction

Reply:
Maybe is good to define the target inflation `I_t` and the base inflation `I_b` and then compute `C = I_t - I_b`: I think it'll be more clear.

Reply:
For now I would stick with the current representation, where the _total_ inflation `I_t` is variable, based on the current staking rate `s_c`.
This is closer to the "theoretical" description.

Reply:
:+1: